### PR TITLE
Simplify pull and configuration

### LINKED ISSUE #33: Simplify remote configuration, ssh key generation/configuration
Following best practices for setting up deploy keys - it's a little challenging to get the remote configuration "just right" at the moment. I think this boils down to:
* Showing the public key in the settings UI
* (Perhaps) automatically generating a key pair for use as a deploy key (need to consider security ramifications of this first though)
* Supporting remote configuration through the main UI

### LINKED ISSUE #229: In Git Project Settings, consistently show green check for valid paths
It's confusing to only have it for (e.g.) the private key file and not the git path / local temp folder path. (Might only be missing when git path is missing.)

### LINKED ISSUE #234: Add support for "git clone" to initialize namespace
That is, as a menu item parallel to Initialize (and available only under the same circumstances)

### LINKED ISSUE #237: Streamline configuration to get ownership right on folders + support git clone
For Linux especially, but also generally helpful: d ##class(SourceControl.Git.API).Configure() should call git APIs over REST to create the folder (acting as irisusr), and also should be able to run git clone (related to #234 with the additional requirement of running it through Configure()).

### LINKED ISSUE #254: Favorite configuration should force logical mode
Reusing the "favorite" approach recently introduced, I found an issue that was also reported from a user via email where the link URL was prefixed with control characters. Solution is to force logical mode for the SQL query in question - e.g.:

```
    set statement = ##class(%SQL.Statement).%New()
    set statement.%SelectMode = 0
    do ##class(%SQL.Statement).%ExecDirect(statement,sql,caption,link).%Display()
```

### PR DESCRIPTION
**Pull request by isc-tleavitt**
(Details in changelog.)

Fixes #234
Fixes #237
Fixes #33
Fixes #254
Fixes #229

### CODE REVIEW

**Comment by isc-tleavitt** [comment]
@isc-pbarton this is a working draft... scope crept a bit. :)

File: csp/socket.csp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        Do ##class(SourceControl.Git.Utils).Clone(remote)
+		Write !,"Done."
+    } ElseIf %request.Get("method") = "sshkeygen" {
+        Do ##class(SourceControl.Git.Utils).GenerateSSHKeyPair()
+		Write !,"Done."
+    } Else {
+        Write !!,"Invalid method selected.",!!
+    }
+</script>
+<script method="OnPreServer" language="cache" returntype="%Status" runat="server" classmethod="##(+$Piece($STACK($STACK-3,"PLACE"),"+",2)<10)##">
```
Comment (isc-tleavitt) [inline csp/socket.csp:21]: I'd really rather not do this... probably just need to bite the bullet and add a %z class to the package.

File: csp/socket.csp [hunk 2]
```diff
@@ -0,0 +1,170 @@
+<csp:class super="%CSP.WebSocket" abstract="1">
```
Comment (isc-tleavitt) [inline csp/socket.csp:1]: Shouldn't actually have abstract

**Comment by isc-tleavitt** [comment]
@isc-pbarton the last bit of work I intend for this is done, good to review now.

File: cls/SourceControl/Git/Settings.cls [hunk 3]
```diff
@@ -104,8 +104,64 @@ ClassMethod Configure() As %Boolean [ CodeMode = objectgenerator ]
...
+{
+    set defaultPromptFlag = $$$DisableBackupCharMask + $$$TrapCtrlCMask + $$$EnableQuitCharMask + $$$DisableHelpCharMask + $$$DisableHelpContextCharMask + $$$TrapErrorMask
+    if (..privateKeyFile '= "") && '##class(%File).Exists(..privateKeyFile) {
+        set value = 1
+        set response = ##class(%Library.Prompt).GetYesNo("Do you wish to create a new SSH key pair?",.value,,defaultPromptFlag)
+        if (response '= $$$SuccessResponse) {
+            quit
+        }
+        if value {
+            #dim workMgr As %SYSTEM.AbstractWorkMgr
```
Comment (isc-pbarton) [inline cls/SourceControl/Git/Settings.cls:121]:
```suggestion
            // using work queue manager ensures proper OS user context when running ssh-keygen
            #dim workMgr As %SYSTEM.AbstractWorkMgr
```

**Review by isc-pbarton — COMMENTED** [review]
All looks good to me! Just added one suggestion, a comment explaining what the work queue manager is for here.

**Review by isc-pbarton — APPROVED** [review]

**Pull request merged**